File: src/main/java/com/tsm/atelier/domain/auth/repository/RefreshTokenRepository.java
package com.tsm.atelier.domain.auth.repository;

import com.tsm.atelier.domain.auth.RefreshToken;
import java.util.Optional;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;

public interface RefreshTokenRepository extends JpaRepository<RefreshToken, Long> {
  Optional<RefreshToken> findByTokenHash(String tokenHash);

  @Modifying
  @Query("UPDATE RefreshToken r SET r.revoked = true WHERE r.user.id = :userId")
  void revokeAllByUserId(UUID userId);

  @Modifying
  @Query(
      "UPDATE RefreshToken r SET r.revoked = true WHERE r.familyId = :familyId AND r.revoked = false")
  void revokeFamily(UUID familyId);
}

File: src/main/kotlin/com/tm/tsmatelier/core/repository/RefreshTokenRepository.kt
package com.tm.tsmatelier.core.repository

import com.tm.tsmatelier.core.entity.RefreshTokenEntity
import com.tm.tsmatelier.core.entity.UserEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import java.util.Optional
import java.util.UUID

interface RefreshTokenRepository : JpaRepository<RefreshTokenEntity, UUID> {
    fun findByToken(token: String): Optional<RefreshTokenEntity>

    @Query("SELECT t FROM RefreshTokenEntity t WHERE t.user = :user AND t.revoked = false")
    fun findAllValidTokensByUser(user: UserEntity): List<RefreshTokenEntity>

    fun findByUser(user: UserEntity): Optional<RefreshTokenEntity>
}
